docs/en/developer/udf_develop_guide.md

```diff
@@ -0,0 +1,152 @@
+# UDF Function Development Guideline
+## 1. Background
+Although there already are hundreds of built-in functions, they can not satisfy the needs in some cases. In the past, this could only be done by developing new built-in functions. Built-in function development requires a relatively long cycle because it needs to recompile binary files and wait for new version release.
+In order to help users quickly and flexibly resolve specific feature computing requirements, we develop the mechanism of user dynamic registration function.
+
+SQL functions can be divided into single-line functions and aggregate functions. At present, only the custom development of single-line functions is supported. An introduction to single-line functions and aggregate functions can be seen [here](./built_in_function_develop_guide.md).
+## 2. Development Procedures
+### 2.1 Develop UDF functions
+#### 2.1.1 Naming Specification of C++ Built-in Function
+- The naming of C++ built-in function should follow the [snake_case](https://en.wikipedia.org/wiki/Snake_case) style.
+- The name should clearly express the function's purpose.
+- The name of a function should not be the same as the name of a built-in function or other custom functions. The list of all built-in functions can be seen [here](../reference/sql/functions_and_operators/Files/udfs_8h.md).
+
+#### 2.1.2 
+The types of the built-in C++ functions' parameters should be BOOL, NUMBER, TIMESTAMP, DATE, or STRING.
+The SQL types corresponding to C++ types are shown as follows:
+
+| SQL Type  | C/C++ Type  |
+|:----------|:------------|
+| BOOL      | `bool`      |
+| SMALLINT  | `int16_t`   |
+| INT       | `int32_t`   |
+| BIGINT    | `int64_t`   |
+| FLOAT     | `float`     |
+| DOUBLE    | `double`    |
+| STRING    | `StringRef` |
+| TIMESTAMP | `Timestamp` |
+| DATE      | `Date`      |
+
+
+#### 2.1.3 函数参数和返回值
+
+**Return Value**:
+
+* If the output type of the UDF is basic type, it will be processed as return value.
+* If the output type of the UDF is STRING, TIMESTAMP or DATE, it will return through the last parameter of the function.
+
+**Parameters**: 
+
+* If the parameter is basic type, it will be passed as value. 
+* If the output type of the UDF is STRING, TIMESTAMP or DATE, it will be passed by pointer. 
+* The first parameter must be `UDFContext* ctx`. The definition of [UDFContext](../../../include/udf/openmldb_udf.h) is:
+
+```c++
+    struct UDFContext {
+        ByteMemoryPool* pool;  // 用来分配内存
```

Comment:
translate the Chinese comments in the entire doc

Reply:
The translation is finished. I am sorry I have forgoten it.